index.html

```diff
@@ -411,43 +501,103 @@ <h2>
-                <strong><dfn data-dfn-for="BatteryManager"><code id=
-                "widl-BatteryManager-onchargingchange">onchargingchange</code></dfn></strong>
+                <dfn>onchargingchange</dfn>
               </td>
               <td>
-                <code><dfn>chargingchange</dfn></code>
+                `chargingchange`
               </td>
             </tr>
             <tr>
               <td>
-                <strong><dfn data-dfn-for="BatteryManager"><code id=
-                "widl-BatteryManager-onchargingtimechange">onchargingtimechange</code></dfn></strong>
+                <dfn>onchargingtimechange</dfn>
               </td>
               <td>
-                <code><dfn>chargingtimechange</dfn></code>
+                `chargingtimechange`
               </td>
             </tr>
             <tr>
               <td>
-                <strong><dfn data-dfn-for="BatteryManager"><code id=
-                "widl-BatteryManager-ondischargingtimechange">ondischargingtimechange</code></dfn></strong>
+                <dfn>ondischargingtimechange</dfn>
               </td>
               <td>
-                <code><dfn>dischargingtimechange</dfn></code>
+                `dischargingtimechange`
               </td>
             </tr>
             <tr>
               <td>
-                <strong><dfn data-dfn-for="BatteryManager"><code id=
-                "widl-BatteryManager-onlevelchange">onlevelchange</code></dfn></strong>
+                <dfn>onlevelchange</dfn>
               </td>
               <td>
-                <code><dfn>levelchange</dfn></code>
+                `levelchange`
               </td>
             </tr>
           </tbody>
         </table>
       </section>
+      <section>
+        <h3>
+          Algorithms
+        </h3>
+        <p>
+          To <dfn>update the battery status and notify</dfn> given an internal
+          slot |slot|, a |value| and an |eventName|, run the following steps:
+        </p>
+        <ol class="algorithm">
+          <li>Let |global| be the <a>current global object</a>.
+          </li>
+          <li>If |global| is not a {{Window}}, abort these steps.
+          </li>
+          <li>Let |navigator:Navigator| be |global|'s associated {{Navigator}}.
+          </li>
+          <li>Let |batteryManager:BatteryManager| be the value of |navigator|.
+          {{Navigator/[[BatteryManager]]}}.
+          </li>
+          <li>If |batteryManager| is `null`, abort these steps.
+          </li>
+          <li>
+            <a>Queue a global task</a> on the <a>battery status task source</a>
+            given |global| to run the following steps:
+            <ol>
+              <li>Set |batteryManager|.|slot| to |value|.
+              </li>
+              <li>
+                <a>Fire an event</a> named |eventName| at |batteryManager|.
+              </li>
+            </ol>
+          </li>
+        </ol>
+      </section>
+      <section id="multiple-batteries">
```

Comment:
As a followup, I wonder if this should be actually integrated into the definition of each slot. That way, it's all kept together.

Single battery, do x... 

Multiple batteries, do y...